libs/docs/core/rtl-service/examples/rtl-service-basic-example/rtl-service-basic-example.component.ts

```diff
@@ -0,0 +1,50 @@
+import { Direction } from '@angular/cdk/bidi';
+import { ChangeDetectionStrategy, Component, computed, effect, inject } from '@angular/core';
+import { FormsModule } from '@angular/forms';
+import { ButtonComponent, FormLabelComponent, RtlService, SwitchComponent, TextComponent } from '@fundamental-ngx/core';
+
+@Component({
+    selector: 'fd-rtl-service-basic-example',
+    templateUrl: 'rtl-service-basic-example.component.html',
+    imports: [TextComponent, ButtonComponent, FormLabelComponent, SwitchComponent, FormsModule],
+    changeDetection: ChangeDetectionStrategy.OnPush,
+    standalone: true
+})
+export class RtlServiceBasicExampleComponent {
+    // Injecting the RtlService
+    private _rtlService = inject(RtlService);
+
+    // Example text
+    text = `
+    Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et
+    dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut
+    aliquip ex ea commodo consequat.
+
+    Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore
+    eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia
+    deserunt mollit anim id est laborum.`;
+
+    // Signal to track RTL state
+    rtl$ = computed(() => !!this._rtlService?.rtlSignal());
+    // Signal to track a direction
+    _dir$ = computed<Direction>(() => {
+        return this.rtl$() ? 'rtl' : 'ltr';
+    });
+
+    constructor() {
+        effect(() => {
+            this._updateTextContainerDirection();
```

Comment:
I am new to signals so forgive me (still doing most of my work in Angular 15). I am confused why there would be an effect here, what are you demonstrating?

Reply:
If you look at the `_updateTextContainerDirection` function, it uses a signal within the function. I'm not directly using the signals in the effect; instead, `_updateTextContainerDirection` handles it. When the `_dir$()` signal changes, the effect will work.

Reply:
So doesn't clicking the toggle button call `_updateTextContainerDirection()`, which triggers the effect, which calls `_updateTextContainerDirection()` again? Is that intended?

Reply:
You're absolutely correct. There's no need to call `_updateTextContainerDirection` explicitly on button click. The effect will automatically trigger whenever `_dir$()` changes.